src/buffer.h

```diff
@@ -45,17 +45,37 @@ struct mpd_buffer {
 	unsigned read;
 
 	/** the actual buffer */
-	unsigned char data[4096];
+	unsigned char *data;
+
+	/** size of buffer */
+	size_t data_len;
```

Comment:
Badly named variable. This is not the "length" of "data". There is rarely ever that much data in the buffer!

Reply:
You are right. I've renamed to `data_size`. Fixed in https://github.com/MusicPlayerDaemon/libmpdclient/pull/53/commits/3cf928540d98463e2cebfa51b42f7a929e63135f

Reply:
But that's still a bad name. Your data is not this size. You don't have that much data.

Reply:
Could we call it `data_max_size` then?

Reply:
That's still ambiguous - reading such a name, I'd think that's the configured limit. What about `capacity`?

Reply:
changed in https://github.com/MusicPlayerDaemon/libmpdclient/commit/ef706f9e741c8778429b0a99776b8a35625da943